test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -4087,10 +4089,22 @@ public async Task Restore_WithHttpSource_Warns()
             // Assert
             result.Success.Should().BeTrue(because: logger.ShowMessages());
             result.LockFile.Libraries.Should().HaveCount(0);
-            result.LockFile.LogMessages.Should().HaveCount(1);
-            IAssetsLogMessage logMessage = result.LockFile.LogMessages[0];
-            logMessage.Code.Should().Be(NuGetLogCode.NU1803);
-            logMessage.Message.Should().Be("You are running the 'restore' operation with an 'HTTP' source, 'http://api.source/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.");
+
+            string expectedWarning = "You are running the 'restore' operation with an 'HTTP' source, 'http://api.source/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.";
+            string unExpectedWarning = "You are running the 'restore' operation with an 'HTTP' source, 'https://api.source/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.";
+
+            if (isHttpWarningExpected)
+            {
+                result.LockFile.LogMessages.Should().HaveCount(1);
+                IAssetsLogMessage logMessage = result.LockFile.LogMessages[0];
+                logMessage.Code.Should().Be(NuGetLogCode.NU1803);
+                Assert.Contains(expectedWarning, logMessage.Message);
+                Assert.DoesNotContain(unExpectedWarning, logMessage.Message);
```

Comment:
I like that this test is thorough, but is it clearer to just assert `Equals` on the expected warning, then drop the unexpected warning completely? 

```suggestion
```

Reply:
If you mean why not do `Assert.Equals("NU1234: This is the expected message")`, then that's high risk of causing problems for people who wish to contribute to NuGet but don't use English as their locale: https://github.com/NuGet/Home/issues/12820. The risk will significantly increase if we eventually do localization more like the rest of github.com/dotnet repos, since local builds can have the satellite assemblies.

Reply:
> If you mean why not do Assert.Equals("NU1234: This is the expected message")

No, I do not mean that. I expect we would continue using the localized variable.

Reply:
@jeffkl requested my review, but after looking at the latest commits, and this thread, this hasn't been addressed.

Reply:
Thanks for reviewing! Sorry for taking long to address this. This is fixed.